src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -1285,4 +1287,85 @@ public JiraSite build() {
         }
     }
 
+    // helper methods
+    // yes this class hierarchy can be a real big mess...
+
+    /**
+     *
+     * @param item the Jenkins {@link Item} can be a {@link Job} or {@link Folder}
+     * @return the parent as {@link ItemGroup} which can be {@link Jenkins} or {@link Folder}
+     */
+    public static ItemGroup map(Item item){
+        ItemGroup parent = null;
+        if (item != null){
+            parent = item instanceof Folder ? (Folder) item : item.getParent();
+        }
+        return parent;
+    }
+
+    public static List<JiraSite> getJiraSites(Item item){
+        ItemGroup itemGroup = JiraSite.map(item);
+        List<JiraSite> sites = (itemGroup instanceof Folder) ?
+            getSitesFromFolders(itemGroup) : JiraGlobalConfiguration.get().getSites();
+        return sites;
+    }
+
+    public static List<JiraSite> getSitesFromFolders(ItemGroup itemGroup) {
+        List<JiraSite> result = new ArrayList<>();
+        while (itemGroup instanceof AbstractFolder<?> ) {
+            AbstractFolder<?> folder = (AbstractFolder<?>) itemGroup;
+            JiraFolderProperty jiraFolderProperty = folder.getProperties()
+                .get(JiraFolderProperty.class);
+            if (jiraFolderProperty != null && jiraFolderProperty.getSites().length != 0) {
+                result.addAll( Arrays.asList( jiraFolderProperty.getSites()));
+            }
+            itemGroup = folder.getParent();
+        }
+        return result;
+    }
+
+
+    /**
+     * Gets the effective {@link JiraSite} associated with the given project.
+     *
+     * @return <code>null</code> if no such was found.
+     */
+    public static JiraSite get(Job<?, ?> p) {
```

Comment:
`@Nullable`?

Reply:
well TBH I'm not fan of this as it's documented in the javadoc but doesn't hurt for sure :)